src/components/Map.js

```diff
@@ -0,0 +1,62 @@
+import GoogleMap from 'google-map-react'
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import Cluster from './Cluster'
+import Location from './Location'
+
+const Map = (props) => {
+  const {
+    bootstrapURLKeys,
+    view,
+    markerData,
+    showLocations,
+    onZoomAnimationEnd,
+    onDragEnd,
+    onClusterClick,
+    onLocationClick,
+  } = props
+
+  return (
+    <GoogleMap
+      bootstrapURLKeys={bootstrapURLKeys}
+      center={view.center}
+      zoom={view.zoom}
+      onZoomAnimationEnd={onZoomAnimationEnd}
+      onDragEnd={onDragEnd}
+    >
+      {markerData.map((marker, index) =>
+        showLocations ? (
+          <Location
+            key={marker.id}
+            onClick={onLocationClick}
+            id={marker.id}
+            lat={marker.lat}
+            lng={marker.lng}
+          />
+        ) : (
+          <Cluster
+            key={index}
+            onClick={onClusterClick}
+            lat={marker.lat}
+            lng={marker.lng}
+            count={marker.count}
+          />
+        ),
+      )}
+    </GoogleMap>
+  )
+}
+
+Map.propTypes = {
+  bootstrapURLKeys: PropTypes.object.isRequired,
```

Comment:
It may be best to expose only an `apiKey` prop here, rather than the entire `bootstrapURLKeys` object.